WormsApp/board.cpp

```diff
@@ -14,13 +13,13 @@ Board::Board(int width, int height) : nextId_{1}, killAll_{false} {
 }
 
 Board::~Board() {
- for(auto& w: getWorms()){
-     w.second.join();}
- }
-
+    for(auto& m: std::move(worms_.getMapper())){
```

Comment:
I wouldn't do `std::move` here. I'm not sure how these objects will be destroyed because of this. Your wrapper has private map which will be destroyed when the wrapper is destroyed, while here you are stealing its reference and you are invalidating it with `std::move`.

I'd stick with:
```
for (auto& m : worms_.getMapper()) {
  m.second.join();
}
```

I'd also do `getMapper` method return a const reference, but probably compiler will not allow for that. You may give it a try.